src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -98,79 +149,242 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
         return indexModel;
     }
 
-    // Minimal
+    /**
+     * Overload for creating a new pod index with environment and podType.
```

Comment:
Just to make things super clear, I'd make this also say `"....and podType, the minimum required parameters for building a pods index."`